HIP/hip-100.md

```diff
@@ -0,0 +1,479 @@
+
+---
+
+##### `getLedgerId` (): `LedgerId`
+
+Return the ledger ID
+
+---
+
+##### `getAccountId` (): `AccountID`
+
+Return the account ID associated with this signer
+
+---
+
+##### `async getAccountBalance` (): `AccountBalance`
+
+Fetch the account's balance
+
+---
+
+##### `async getAccountInfo` (): `AccountInfo`
+
+Fetch the account's info
+
+---
+
+##### `async getTransactionRecords` (): `List` < `TransactionRecord` >
+
+Fetch the last transaction records for this account using `TransactionRecordQuery`
+
+---
+
+##### `async getTransactionRecords` (): `List` < `TransactionRecord` >
+
+Fetch the last transaction records for this account using `TransactionRecordQuery`
+
+---
+</details>
+
+<details>
+<summary>abstract class Provider</summary>
+
+> abstract class `Provider`
+
+### Methods
+
+##### `getLedgerId` (): `LedgerId`
+
+Return the ID of the current network
+
+---
+
+##### `async getAccountBalance` ( `accountId`: `AccountId` ): `AccountBalance`
+
+Get the balance for an account
+
+---
+
+##### `async getAccountInfo` ( `accountId`: `AccountId` ): `AccountInfo`
+
+Get the info for an account
+
+---
+
+##### `async getTransactionReceipt` ( `transactionId`: `TransactionId` ): `TransacitonReceipt`
+
+Get a receipt for a transaction ID
+
+---
+
+##### `async sendTransaction` ( `transaction`: `Transaction` ): `TransactionResponse`
+
+Sign and send a transaction using the wallet
+
+---
+
+##### `async waitForReceipt` ( `response`: `TransactionResponse` ): `TransactionReceipt`
+
+Wait for the receipt for a transaction response
+
+**NOTE**: This is different than `getTransactionReceipt()` this method requires a `nodeId` which
+is set inside `TransactionResponse` and as a result should not be able to fail with `RECEIPT_NOT_FOUND`
+
+---
+</details>
+
+<details>
+<summary>abstract class Wallet extends Signer</summary>
+
+### Static Methods
+
+##### `withPrivateKey` ( `privateKey`: `PrivateKey` ): `Wallet`
+
+Create a wallet using a private key
+
+---
+
+### Methods
+
+##### `getProvider` (): `Provider`
+
+Return the provider
+
+---
+
+##### `getPublicKey` (): `PublicKey`
+
+Return the public key associated with this wallet.
+
+---
+
+##### `async createRandomED25519` (): `Wallet`
+
+Creates a wallet with a new ED25519 key
+
+**NOTE**: This would create an alias key account on Hedera
+
+---
+
+##### `async createRandomECDSA` (): `Wallet`
+
+Creates a wallet with a new ECDSA key
+
+**NOTE**: This would create an alias key account on Hedera
+
+---
+</details>
+
+<details>
+<summary>class SignerSignature</summary>
+
+### Fields
+
+##### `publicKey`: `PublicKey`
+
+The public key that signed this request
+
+##### `signature`: `bytes`
+
+The siganture for the message
+
+</details>
+
+<details>
+<summary>class LocalWallet implements Wallet</summary>
+
+### Static Methods
+
+### Constructor
+
+##### `constructor`()
+
+Creates an `LocalWallet` from the environment variables `OPERATOR_KEY`, `OPERATOR_ID`, and `HEDERA_NEWTORK`
```

Comment:
Can we get a variant with explicit args as well?

Reply:
Not sure we should, if someone wants to do this, they should create their own implementation of `Wallet` and use that as a Provider/Signer. This would ensure the provider/signer pattern is adhered to.

Reply:
I agree with @gregscullard here. I think the base `LocalWallet` that is included in the SDK should use hard coded environment variables. Perhaps we can change these required ones to be `HEDERA_OPERATOR_KEY` and `HEDERA_OPERATOR_ID` to make it less likely these envs collide with other envs users might be using?